bigquery_etl/util/table_filter.py

```diff
@@ -41,14 +40,22 @@ def glob_predicate(table: str, pattern: re.Pattern, arg: str) -> bool:
     if arg == "except":
         matched = not matched
     if not matched:
-        logging.debug(f"Skipping {table} due to --{arg} argument")
+        logging.info(f"Skipping {table} due to --{arg} argument")
     return matched
 
 
 def get_table_filter(args: Namespace) -> Callable[[str], bool]:
     """Get a function that evaluates whether a given table that should be included."""
-    for patterns, arg in [(args.only_tables, "only"), (args.except_tables, "except")]:
-        if patterns:
-            pattern = compile_glob_patterns(patterns)
-            return partial(glob_predicate, pattern=pattern, arg=arg)
+    predicates = [
+        partial(glob_predicate, pattern=compile_glob_patterns(patterns), arg=arg)
+        for patterns, arg in [
+            (args.only_tables, "only"),
+            (args.except_tables, "except"),
+        ]
+        if patterns
+    ]
+    if len(predicates) > 1:
+        return lambda table: all(predicate(table) for predicate in predicates)
+    if len(predicates) == 1:
+        return predicates[0]
     return lambda table: True
```

Comment:
```suggestion
    return lambda table: all(predicate(table) for predicate in predicates)
```

I don' think the if conditions are necessary here, as the `all` solution will be valid even down to the empty list case `all([]) == True`.